### PR TITLE
Add support for custom working directory for puppet

### PR DESCRIPTION
**Pull request by ColinHebert**
Similar behaviour to Vagrant's support of masterless puppet (see `Configuring Hiera` section):

https://docs.vagrantup.com/v2/provisioning/puppet_apply.html

### CODE REVIEW

**Comment by ColinHebert** [comment]
Any feedback on this PR?

**Comment by ColinHebert** [comment]
Any news?

File: provisioner/puppet-masterless/provisioner.go [hunk 1]
```diff
@@ -73,7 +78,8 @@ func (p *Provisioner) Prepare(raws ...interface{}) error {
 
 	// Set some defaults
 	if p.config.ExecuteCommand == "" {
-		p.config.ExecuteCommand = "{{.FacterVars}} {{if .Sudo}} sudo -E {{end}}" +
+		p.config.ExecuteCommand = "cd {{.WorkingDir}} && " +
```
Comment (sethvargo) [inline provisioner/puppet-masterless/provisioner.go]: It seems like using `cd` here is not a very cross-platform (especially for Windows). I believe there is a way to set the working directory when we exec the command from Go.

**Comment by sethvargo** [comment]
It would be really great to see a test or two for this :smile:

File: provisioner/puppet-masterless/provisioner.go (hunk 1 above)
Comment (ColinHebert) [inline provisioner/puppet-masterless/provisioner.go]: `sudo` would probably be the biggest problem. Not `cd` which is both on Windows and Linux.

**Comment by ColinHebert** [comment]
Fair point for the tests. I actually tried to write one when I open the PR, based on some examples that would be provided in the already existing test file, but I couldn't make out how those tests are working. So rather than having to improve tests, it seems that the need here is to define how the entire provisioner should be tested (which I'm not qualified for unfortunately).

File: provisioner/puppet-masterless/provisioner.go (hunk 1 above)
Comment (sethvargo) [inline provisioner/puppet-masterless/provisioner.go]: Hrm yea - I dug through the code and it does not look like there is a way to specify a directory with `RemoteCmd` (which is what gets unwrapped under the hood). Oh well.

File: website/source/docs/provisioners/puppet-masterless.html.markdown [hunk 2]
```diff
@@ -79,12 +79,18 @@ Optional parameters:
   this folder. If the permissions are not correct, use a shell provisioner
   prior to this to configure it properly.
 
+* `working_directory` (string) - This is the directory from which the puppet command
+  will be run. When using hiera with a relative path, this option allows to ensure
+  that he paths are working properly. If not specified, defaults to the value of
```
Comment (sethvargo) [inline website/source/docs/provisioners/puppet-masterless.html.markdown]: that **"the"** paths

Comment (ColinHebert) [inline website/source/docs/provisioners/puppet-masterless.html.markdown]: Fixed.

**Comment by sethvargo** [comment]
LGTM

**Comment by mitchellh** [comment]
LGTM!

**Pull request merged**